[BEAM-3404] Update KinesisIO to use AWS SDK 1.11.255 and KCL 1.8.8

iemejia opened the pull request:
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/BEAM/issues/) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

lukecwik commented on .gitignore:
```diff
@@ -7,6 +7,7 @@
 .gogradle/
 build/
 vendor/
+.gradletasknamecache
```
Add this to the apache rat configuration in the root build.gradle

iemejia replied:
ok

lukecwik commented:
Run Java Gradle PreCommit

iemejia commented:
I just updated KinesisReaderIT because the test was disabled and fixed some dep problem to finally run it with success, Thanks for suggesting this. PTAL @lukecwik

lukecwik commented:
Can you take a look at this test failure: https://builds.apache.org/job/beam_PreCommit_Java_GradleBuild/794/testReport/junit/org.apache.beam.sdk.io.kinesis/KinesisReaderTest/watermarkAdvancesWhenEnoughRecordsReadRecently/

echauchot commented:
LGTM as well

iemejia commented:
Run Java Gradle PreCommit

iemejia commented:
@lukecwik I could not reproduce this issue both locally or when I rerun the gradle tests I am merging the code since seems to work and the issue seems to be only on the gradle build, or maybe is related to another issue with Kinesis flaky tests. https://issues.apache.org/jira/browse/BEAM-3228

I will be taking a look at the jenkins jobs and if I find that it repeats consistently I will create a JIRA for this.
Thanks for the review, and thanks everybody else.
